src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -3192,3 +3192,30 @@ class StatementSegment(BaseSegment):
         Ref("TransactionStatementSegment"),
         Ref("ExecuteScriptSegment"),
     )
+
+
+@exasol_dialect.segment(replace=True)
+class FunctionSegment(BaseSegment):
+    """A scalar or aggregate function.
+
+    Maybe in the future we should distinguish between
+    aggregate functions and other functions. For now
+    we treat them the same because they look the same
+    for our purposes.
+    """
+
+    type = "function"
+    match_grammar = Sequence(
+        Sequence(
+            Ref("FunctionNameSegment"),
+            Bracketed(
+                Ref(
+                    "FunctionContentsGrammar",
+                    # The brackets might be empty for some functions...
+                    optional=True,
+                    ephemeral_name="FunctionContentsGrammar",
+                )
+            ),
+        ),
+        Ref("PostFunctionGrammar", optional=True),
+    )
```

Comment:
Is this only hear because Exasol doesn't have the DATEADD function? Cause it looks like it has ADD_DAYS which seems to be similar: https://docs.exasol.com/sql_references/functions/alphabeticallistfunctions/add_days.htm

So are we better overriding `DateAddFunctionNameSegment` with that and leaving the `FunctionSegment` alone?

Reply:
I added in logic for ADD_DAYS.  It's not exactly the same since it doesn't include the datepart parameter, so I was not able to just leave the base FunctionSegment alone, but it highlights how the logic can be tweaked by dialect.